File: golang/internal/wow/server/settable_config.go
package server

// SettableConfigKeys are the only ReaderConfig fields writable via /rest/setConfig.
// Mirrors client.DesiredConfig — export-shaping knobs the converter syncs before export.
var SettableConfigKeys = map[string]struct{}{
	"copyMode":                   {},
	"listfileShowFileDataIDs":    {},
	"enableM2Skins":              {},
	"enableSharedTextures":       {},
	"enableSharedChildren":       {},
	"enableAbsoluteMTLPaths":     {},
	"enableAbsoluteCSVPaths":     {},
	"removePathSpaces":           {},
	"removePathSpacesCopy":       {},
	"exportTextureFormat":        {},
	"exportModelFormat":          {},
	"exportM2Bones":              {},
	"exportM2Meta":               {},
	"exportWMOMeta":              {},
	"modelsExportSkin":           {},
	"modelsExportSkel":           {},
	"modelsExportBone":           {},
	"modelsExportAnim":           {},
	"modelsExportUV2":            {},
	"modelsExportTextures":       {},
	"modelsExportAlpha":          {},
	"modelsExportAnimations":     {},
	"modelsExportCollision":      {},
}

// IsSettableConfigKey reports whether key may be changed over HTTP.
func IsSettableConfigKey(key string) bool {
	_, ok := SettableConfigKeys[key]
	return ok
}

File: src/lib/wow/server/settable-config.ts
import type { WowReaderConfig } from './config';

/** Keys the converter may sync via /rest/setConfig (mirrors wow-data-client desiredConfig). */
export const SETTABLE_CONFIG_KEYS = new Set<keyof WowReaderConfig>([
  'copyMode',
  'listfileShowFileDataIDs',
  'enableM2Skins',
  'enableSharedTextures',
  'enableSharedChildren',
  'enableAbsoluteMTLPaths',
  'enableAbsoluteCSVPaths',
  'removePathSpaces',
  'removePathSpacesCopy',
  'exportTextureFormat',
  'exportModelFormat',
  'exportM2Bones',
  'exportM2Meta',
  'exportWMOMeta',
  'modelsExportSkin',
  'modelsExportSkel',
  'modelsExportBone',
  'modelsExportAnim',
  'modelsExportUV2',
  'modelsExportTextures',
  'modelsExportAlpha',
  'modelsExportAnimations',
  'modelsExportCollision',
]);

export function isSettableConfigKey(key: string): key is keyof WowReaderConfig {
  return SETTABLE_CONFIG_KEYS.has(key as keyof WowReaderConfig);
}
